### PR TITLE
postgres_mixin: jsonnetify mixin

### PR DESCRIPTION
**Pull request by paulfantom**
Main motivation behind this PR is to allow parametrization of alert expressions by adding job selector. However to to this, mixin was refactored to bring it closer to ones from https://github.com/prometheus/prometheus and https://github.com/kubernetes-monitoring/kubernetes-mixin/. Notably following things were done:
- alerts file was rewritten from YAML to jsonnet
- files structure was changed (added `alerts.libsonnet` and `dashboards.libsonnet`)
- promQL expressions were rewritten to multiline for readability
- promQL expressions were rewritten from `sum (EXPRESSION) by (SELECTOR)` to `sum by (SELECTOR) (EXPRESSION)` to further improve readability of long queries
- `postgresExporterSelector` selector was added to allow using mixin in multi-tenant environment

Note: Merging this PR renders https://github.com/prometheus-community/postgres_exporter/pull/630 obsolete.

Signed-off-by: Paweł Krupa (paulfantom) <pawel@krupa.net.pl>

### CODE REVIEW

**Review by SuperQ — APPROVED** [review]
Nice, thanks!

File: postgres_mixin/alerts/postgres.libsonnet [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+              description: 'PostgreSQL high number of queries per second on {{ $labels.cluster }} for database {{ $labels.datname }} with a value of {{ $value }}',
+              summary: 'PostgreSQL high number of queries per second {{ $labels.cluster }} for database {{ $labels.datname }}',
+            },
+            expr: |||
+              avg by (datname) (
+                irate(
+                  pg_stat_database_xact_commit{datname!~"template.*",%(postgresExporterSelector)s}[5m]
+                )
+                +
+                irate(
```
Comment (sysadmind) [inline postgres_mixin/alerts/postgres.libsonnet:87]: I don't think these are supposed to be irate, but that can be fixed later.

File: postgres_mixin/alerts/postgres.libsonnet [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+                  pg_stat_database_xact_rollback{datname!~"template.*",%(postgresExporterSelector)s}[5m]
+                )
+              ) > 10000
+            ||| % $._config,
+            'for': '5m',
+            labels: {
```
Comment (sysadmind) [inline postgres_mixin/alerts/postgres.libsonnet:93]: We will need to add an entry to the changelog that the labels changed.

**Review by sysadmind — APPROVED** [review]
LGTM. Thanks!

File: postgres_mixin/alerts/postgres.libsonnet [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+            annotations: {
+              description: 'PostgreSQL high number of slow queries {{ $labels.cluster }} for database {{ $labels.datname }} with a value of {{ $value }} ',
+              summary: 'PostgreSQL high number of slow on {{ $labels.cluster }} for database {{ $labels.datname }} ',
+            },
+            expr: |||
+              avg(
+                rate by (datname) (
+                  pg_stat_activity_max_tx_duration{datname!~"template.*",%(postgresExporterSelector)s}[2m]
+                )
+              ) > 2 * 60
```
Comment (friedrich-at-adobe) [inline postgres_mixin/alerts/postgres.libsonnet:68]: the previous expression was 
```
avg(
  rate(
    pg_stat_activity_max_tx_duration{datname!~"template.*"}[2m]
  )
) by (datname) > 2 * 60
```
the new one doesn't work

Comment (sysadmind) [reply postgres_mixin/alerts/postgres.libsonnet:68]: If you move the `by (datname)` to the line above to read `avg by (datname) (`, does this fix the query? I think the meaningful difference is which function is doing the grouping.

Comment (friedrich-at-adobe) [reply postgres_mixin/alerts/postgres.libsonnet:68]: yes, this expression works too
```
avg by (datname) (
  rate(
    pg_stat_activity_max_tx_duration{datname!~"template.*"}[2m]
  )
)  > 2 * 60
```

**Pull request merged**